src/hera/workflows/_mixins.py

```diff
@@ -898,6 +921,121 @@ def _get_params_from_items(with_items: List[Any]) -> Optional[List[Parameter]]:
     return [Parameter(name=n, value=f"{{{{item.{n}}}}}") for n in with_items[0].keys()]
 
 
+def _model_attr_setter(attrs: List[str], model: BaseModel, value: Any):
```

Comment:
I am slightly confused by the logic here - why is it necessary to check all but the last attr then set the last one only? Also, very picky comment, but `_set_model_attr` seems more aligned with the _action_ this performs